crates/matrix-sdk/tests/integration/encryption/backups.rs

```diff
@@ -574,7 +574,7 @@ async fn setup_create_room_and_send_message_mocks(server: &wiremock::MockServer)
     Mock::given(method("POST"))
         .and(path("/_matrix/client/r0/keys/upload"))
         .and(header("authorization", "Bearer 1234"))
-        .respond_with(ResponseTemplate::new(404).set_body_json(json!({
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
```

Comment:
Reported here, https://github.com/matrix-org/matrix-rust-sdk/pull/2939/files#r1749634419. Reading the diffs, it looks like a copy-pasta error. I think it's fine to return a 200 here.

I am however really surprised it didn't crash more often. Returning a 200 instead of a 404 has also really few impacts. We may consider rewriting these tests but probably not today.